### PR TITLE
add-multi-line-variable-support-for-variables

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**

### CODE REVIEW

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java [hunk 1]
```diff
@@ -639,6 +640,24 @@ private BlurbItem createModelItem(String model) {
         return modelLabel;
     }
 
+    private FormItem getVariableItem(Entry<String, JobVariable> var) {
+        if ("NOT_EMPTY_STRING".equals(var.getValue().getModel()) || "PA:JSON".equals(var.getValue().getModel()) ||
```
Comment (tao-xinxiu) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java:644]: same as above

**Review by tao-xinxiu — APPROVED** [review]

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/ActionsWindow.java [hunk 2]
```diff
@@ -294,6 +295,24 @@ private void initVariablesVisualForm(Map<String, JobVariable> variablesByGroup,
         }
     }
 
+    private FormItem getVariableItem(Map.Entry<String, JobVariable> var) {
+        if ("NOT_EMPTY_STRING".equals(var.getValue().getModel()) || "PA:JSON".equals(var.getValue().getModel()) ||
```
[the remark this answers is not in the document]
Comment (oanaBiancaSchiopu) [reply scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/ActionsWindow.java:299]: I will change it to PA:NOT_EMPTY_STRING
I am not sure if ar.getValue().getModel() always returns the variable modal with all letters in capital, but I can handle lower letter too

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java [hunk 3]
```diff
@@ -639,6 +640,27 @@ private BlurbItem createModelItem(String model) {
...
+            "PA:REGEXP".equalsIgnoreCase(var.getValue().getModel())) {
+            return createVariableTextAreaItem(var);
+        } else {
+            return createVariableItem(var);
+        }
+    }
+
+    private TextAreaItem createVariableTextAreaItem(Entry<String, JobVariable> var) {
+        TextAreaItem t = new TextAreaItem(var.getKey(), var.getKey());
+        t.setValue(var.getValue().getValue().replaceAll("ENC((.*))", "*******"));
```
Comment (fviale) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java:657]: not sure this is hidden. Encrypted values are for model PA:HIDDEN

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java [hunk 4]
```diff
@@ -639,6 +640,27 @@ private BlurbItem createModelItem(String model) {
         return modelLabel;
     }
 
+    private FormItem getVariableItem(Entry<String, JobVariable> var) {
+        if (var.getValue().getModel() == null || "".equals(var.getValue().getModel()) ||
```
Comment (fviale) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java:644]: you can use Strings.isNullOrEmpty()

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/ActionsWindow.java [hunk 5]
```diff
@@ -294,6 +295,27 @@ private void initVariablesVisualForm(Map<String, JobVariable> variablesByGroup,
         }
     }
 
+    private FormItem getVariableItem(Map.Entry<String, JobVariable> var) {
+        if (var.getValue().getModel() == null || "".equals(var.getValue().getModel()) ||
```
Comment (fviale) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/ActionsWindow.java:299]: you can use Strings.isNullOrEmpty()

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/ActionsWindow.java [hunk 6]
```diff
@@ -294,6 +295,27 @@ private void initVariablesVisualForm(Map<String, JobVariable> variablesByGroup,
...
+            "PA:REGEXP".equalsIgnoreCase(var.getValue().getModel())) {
+            return createVariableTextAreaItem(var.getKey(), var.getValue().getValue());
+        } else {
+            return createVariableItem(var.getKey(), var.getValue().getValue());
+        }
+    }
+
+    private TextAreaItem createVariableTextAreaItem(String name, String value) {
+        TextAreaItem t = new TextAreaItem(name, name);
+        t.setValue(value.replaceAll("ENC((.*))", "*******"));
```
Comment (fviale) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/ActionsWindow.java:312]: not sure this is needed. Encrypted values are for model PA:HIDDEN

**Review by fviale — APPROVED** [review]
Minor remarks

**Review by fviale — APPROVED** [review]
Please remember to squash your commits while merging

**Pull request merged**